resource_harbor_shipment_env.go

```diff
@@ -607,6 +621,21 @@ func transformTerraformToShipmentEnvironment(d *schema.ResourceData, existingShi
 		result.EnvVars = copyUserDefinedEnvVars(existingShipmentEnvironment.EnvVars)
 	}
 
+	// annotations
+	annotationsResource := d.Get("annotations")
+	if annotations, ok := annotationsResource.(map[string]interface{}); ok && len(annotations) > 0 {
+		result.Annotations = make([]AnnotationsPayload, 0)
```

Comment:
very minor nit:  preferable to use a slice here rather than constant reallocation of array
```
result.Annotations = []AnnotationsPayload{}
```